docs/source/architecture.rst

```diff
@@ -73,60 +84,40 @@ We recursively parse each of the elements, using their in built grammars.
       segments, and the *expand* step will let each of those segments refine
       the content within them.
 
-   #. Eventually in that recursive operation we reach segments which have no
-      children (raw elements containing a single token), and so the recursion
+   #. During the recursion, segments which have no children are eventually
```

Comment:
Possible alternate wording:

During the recursion, the parser eventually reaches segments which have no children